lua/autorun/client/cl_chip_lister.lua

```diff
@@ -0,0 +1,321 @@
+    INFO_OFFSET_CHIP = -surface.GetTextSize( table.GetKeys( CHIP_COLORS )[1] .. " " .. stringFormat( CPUS_FORMAT, 0 ) .. STR_MICROSECONDS )
+end
+
+local LISTER_ENABLED = CreateClientConVar( "cfc_chiplister_enabled", 1, true, false, "Enables the Expression2/Starfall chip lister." )
+
+local listerEnabled = LISTER_ENABLED:GetBool()
+
+
+include( "cfc_chip_lister/client/cl_hud.lua" )
+
+
+local function colorToHSV( color )
+    return Color( _colorToHSV( color ) )
+end
+
+local function hsvToColor( color )
+    return _hsvToColor( color.r, color.g, color.b )
+end
+
+local function formatCPUs( num )
+    local usageStr = stringFormat( CPUS_FORMAT, num or 0 )
+    local leadStr = ""
+    local numStr = ""
+    local leadCount = 0
+
+    for i = 1, stringLen( usageStr ) do
+        if stringSub( usageStr, i, i ) == "0" then
+            leadStr = leadStr .. "0"
+            leadCount = leadCount + 1
+        else
+            return leadStr, stringSub( usageStr, leadCount + 1 )
+        end
+    end
+
+    return leadStr, ""
+end
+
+local function getTeamColor( ply )
+    if ply == ID_WOLRD or not isValid( ply ) then
+        return COLOR_WORLD
+    end
+
+    return teamGetColor( getTeam( ply ) )
+end
+
+local function fadeColor( color, fadeOverride )
+    local hsv = colorToHSV( color )
+    local off = fadeOverride or COLOR_FADE_OFFSET
+
+    return _hsvToColor( hsv.r + off.r, hsv.g + off.g, hsv.b + off.b )
+end
+
+COLOR_TEXT_FADED = fadeColor( COLOR_TEXT )
+COLOR_MICROS = fadeColor( COLOR_TEXT, COLOR_FADE_MICROS_OFFSET )
+
+
+cvars.AddChangeCallback( "cfc_chiplister_enabled", function( _, old, new )
+    local state = new ~= "0"
+
+    if state == listerEnabled then return end
+
+    listerEnabled = state
+    
+    net.Start( "CFC_ChipLister_SetEnabled" )
+    net.WriteBool( listerEnabled )
+    net.SendToServer()
+
+    if listerEnabled then
+        render.PushRenderTarget( rtChipLister )
+        cam.Start2D()
+
+            surface.SetDrawColor( COLOR_BACKGROUND )
+            surface.DrawRect( 0, 0, SCREEN_SIZE, SCREEN_SIZE )
+
+            draw.SimpleText( STR_WAITING, FONT_NAME, SCREEN_SIZE_HALF, SCREEN_SIZE_HALF, COLOR_TEXT, TEXT_ALIGN_CENTER, TEXT_ALIGN_CENTER )
+
+        cam.End2D()
+        render.PopRenderTarget()
+    else
+        render.PushRenderTarget( rtChipLister )
+        cam.Start2D()
+
+            surface.SetDrawColor( COLOR_BACKGROUND )
+            surface.DrawRect( 0, 0, SCREEN_SIZE, SCREEN_SIZE )
+
+            draw.SimpleText( STR_ENABLE, FONT_NAME, SCREEN_SIZE_HALF, SCREEN_SIZE_HALF, COLOR_TEXT, TEXT_ALIGN_CENTER, TEXT_ALIGN_CENTER )
+
+        cam.End2D()
+        render.PopRenderTarget()
+    end
+end )
+
+
+hook.Add( "InitPostEntity", "CFC_ChipLister_InformServerOfPlayerChoice", function()
+    timer.Simple( 10, function()
+        net.Start( "CFC_ChipLister_SetEnabled" )
+        net.WriteBool( LISTER_ENABLED:GetBool() )
```

Comment:
Server can get bools from the player https://wiki.facepunch.com/gmod/Player:GetInfo, no need to network it through glua

Reply:
There's no hook or the like on the server for notifying when a player changes a UserInfo convar, so we'd have to determine all the enabled players every time we update the screen, which is not ideal.

Reply:
I'm not sure what you're talking about. All I'm talking about is that players sending a convar on `InitPostEntity` will cause issues, net is not reliable yet at that point (speaking from experience). This is literally only needed on join. I am not talking about any of the other checks I am literally just talking about this specific hook I reacted to.

Reply:
`InitPostEntity` is when the player is able to reliably handle net messages, that's a critical part of its whole use-case in general. You're thinking about `PlayerInitialSpawn`

Reply:
No I am not, `InitPostEntity` isn't reliable.

Reply:
I agree, I think `GetInfoNum` is the way to go here. We can entirely remove this net message and use an inbuilt method of getting a player's preferences. Less code is always better 👍